ts/nix.ts

```diff
@@ -174,6 +196,72 @@ export function toHumanReadable(nixExpr: NixExpression): string {
   return renderInterpolatedString(nixExpr.str, toHumanReadable);
 }
 
+function collectFlakeDeps(nixExpr: NixExpression): Record<string, FlakeDep> {
+  const collect = (arr: Array<NixExpression>) =>
+    arr.reduce(
+      (acc, nixExpr) => ({ ...acc, ...collectFlakeDeps(nixExpr) }),
```

Comment:
Should we throw here if there's duplicate keys that don't map to the same exact flake url? That seems like a situation where we want to abort.

Reply:
Yeah this is probably best for now. I was assuming we'd eventually throw in the functions that build up the AST (as we transition away from nixRaw).